client/my-sites/site-settings/publishing-tools/post-by-email.tsx

```diff
@@ -0,0 +1,138 @@
+import isRegeneratingJetpackPostByEmail from 'calypso/state/selectors/is-regenerating-jetpack-post-by-email';
+import isSiteAutomatedTransfer from 'calypso/state/selectors/is-site-automated-transfer';
+import { isJetpackSite } from 'calypso/state/sites/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+type PostByEmailSettingProps = {
+	isFormPending: boolean;
+	address?: string;
+};
+
+const moduleSlug = 'post-by-email';
+
+export const PostByEmailSetting = ( { isFormPending, address }: PostByEmailSettingProps ) => {
+	const {
+		siteId: selectedSiteId,
+		siteIsJetpack: isJetpack,
+		siteIsAtomic: isAtomic,
+		regenerating: regeneratingPostByEmail,
+		isActive: active,
+		moduleIsUnavailable: moduleUnavailable,
+	} = useSelector( ( state ) => {
+		const siteId = getSelectedSiteId( state ) || 0;
+		const siteIsJetpack = isJetpackSite( state, siteId );
+		const siteIsAtomic = isSiteAutomatedTransfer( state, siteId );
+		const regenerating = isRegeneratingJetpackPostByEmail( state, siteId );
+		const isActive = !! isJetpackModuleActive( state, siteId, moduleSlug );
+		const siteInDevMode = isJetpackSiteInDevelopmentMode( state, siteId );
+		const moduleUnavailableInDevMode = isJetpackModuleUnavailableInDevelopmentMode(
+			state,
+			siteId,
+			moduleSlug
+		);
+
+		return {
+			siteId,
+			siteIsJetpack,
+			siteIsAtomic,
+			regenerating,
+			isActive,
+			moduleIsUnavailable: siteInDevMode && moduleUnavailableInDevMode,
+		};
+	} );
+
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const handleRegenerate = () => {
+		if ( ! selectedSiteId ) {
+			return;
+		}
+
+		const regenerate = regeneratePostByEmail( selectedSiteId );
+
+		dispatch( regenerate );
+	};
+
+	const email = address && address !== 'regenerate' ? address : '';
+	const labelClassName =
+		moduleUnavailable || regeneratingPostByEmail || ! active ? 'is-disabled' : undefined;
+
+	return (
+		<>
+			<FormFieldset>
+				<SupportInfo
+					text={ translate(
+						'Allows you to publish new posts by sending an email to a special address.'
+					) }
+					link={
+						isAtomic
+							? localizeUrl( 'https://wordpress.com/support/post-by-email/' )
+							: 'https://jetpack.com/support/post-by-email/'
+					}
+					privacyLink={ ! isAtomic }
+				/>
+				{ isJetpack ? (
+					<JetpackModuleToggle
+						siteId={ selectedSiteId }
+						moduleSlug="post-by-email"
+						label={ translate( 'Publish posts by sending an email' ) }
+						disabled={ isFormPending || moduleUnavailable }
+					/>
+				) : (
+					<ToggleControl
+						checked={ active }
+						disabled={ isFormPending }
+						label={ translate( 'Post by Email' ) }
+						onChange={ ( checked ) => {
+							if ( ! selectedSiteId ) {
+								return;
+							}
+
+							if ( checked ) {
+								activateModule( selectedSiteId, moduleSlug );
+							} else {
+								deactivateModule( selectedSiteId, moduleSlug );
+							}
```

Comment:
Similar to my other comment about my redux-fu, the fact that this works feels like i've accidentally tapped into some dark magic. Previously these two methods were piped through `connect`, but now they just work and I don't know why. I hate it. Please send help.

Reply:
Your code should not work, since `activateModule ` does nothing, it just returns another function.
To run it in context of redux you need to provide it to `dispatch`, like you did here (https://github.com/Automattic/wp-calypso/pull/90819#discussion_r1609061999)

`dispatch(activateModule( selectedSiteId, moduleSlug ))`

And it's the reason and the answer - why the case with `regeneratePostByEmail ` works well - there is `dispatch` wrapper function.
Hope it answers both your questions 🙂

Reply:
Thank you, that makes more sense, but also didn't at first because the toggle _did_ still work for JP/WoA sites. 

Then it dawned on me: this broken code was never being run. We don't actually need `activateModule` or `deactivateModule` because they're already built into `JetpackModuleToggle`. Putting them in my `onChange` handler didn't impact anything because that handler is only being run for simple sites. Mildly embarrassing, but hey... my handler is less verbose now. I'll call that a win.